File: array_benchmark.nim
import times
import sequtils


proc compute_array(m: any, n: any) =
  var arr = newSeqWith(m, newSeq[int](n))
  for i in 0..m-1:
    for j in 0..n-1:
      arr[i][j] = i*i + j*j  
  echo(arr[m-1][n-1])

var m = 10000
var n = 10000
var n_loops = 5
let time = now()
for i in 0..n_loops-1:
  compute_array(m, n)
echo(now() - time)

File: array_benchmark.go
package main

import (
	"fmt"
	"time"
)

func main() {
	m := 15000
	n := 15000
	n_loop := 5
	start := time.Now()

	for i := 0; i < n_loop; i++ {
		compute_range(m, n)
	}

	duration := time.Since(start)
	fmt.Println(duration)
}

func compute_range(m2 int, n2 int) {
	arr := [15000][15000]int{}
	for j := 0; j < m2; j++ {
		for k := 0; k < n2; k++ {
			arr[j][k] = j*j + k*k
		}
	}
	fmt.Println(arr[15000-1][15000-1])
}
